sentinel-extension/sentinel-datasource-redis/src/test/java/com/alibaba/csp/sentinel/datasource/redis/RedisDataSourceTest.java

```diff
@@ -0,0 +1,111 @@
+package com.alibaba.csp.sentinel.datasource.redis;
+
+import ai.grakn.redismock.RedisServer;
+import com.alibaba.csp.sentinel.datasource.ConfigParser;
+import com.alibaba.csp.sentinel.datasource.DataSource;
+import com.alibaba.csp.sentinel.slots.block.flow.FlowRule;
+import com.alibaba.csp.sentinel.slots.block.flow.FlowRuleManager;
+import com.alibaba.fastjson.JSON;
+import com.alibaba.fastjson.TypeReference;
+import io.lettuce.core.RedisClient;
+import io.lettuce.core.RedisURI;
+import io.lettuce.core.api.sync.RedisCommands;
+import io.lettuce.core.pubsub.StatefulRedisPubSubConnection;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.List;
+
+public class RedisDataSourceTest {
+
+    private static RedisServer server = null;
+
+    private RedisClient client;
+
+    private String ruleKey = "sentinel.flow.rulekey";
+
+    private String channel = "sentinel.flow.channel";
+
+    @Before
+    public void buildResource() {
+        try {
+            //bind to a random port
+            server = RedisServer.newRedisServer();
+            server.start();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        ConfigParser<String, List<FlowRule>> flowConfigParser = buildFlowConfigParser();
+        client = RedisClient.create(RedisURI.create(server.getHost(), server.getBindPort()));
+        initRedisRuleData();
+        DataSource<String, List<FlowRule>> redisDataSource = new RedisDataSource<List<FlowRule>>(client, ruleKey, channel, flowConfigParser);
+        FlowRuleManager.register2Property(redisDataSource.getProperty());
+    }
+
+    @Test
+    public void pub_msg_and_receive_success() {
```

Comment:
The method name should follow camel-case naming style like: `testPubMessageAndReceiveSuccess`.